### PR TITLE
add unit tests for null param to __construct in Router

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

File: test/Router/FastRouteTest.php [hunk 1]
```diff
@@ -33,6 +34,15 @@ public function getRouter()
         );
     }
 
+    public function testGetRouterPropertyWithNullInConstruct()
```
Comment (weierophinney) [inline test/Router/FastRouteTest.php]: Rename to:

```php
public function testWillLazyInstantiateAFastRouteCollectorIfNoneIsProvidedToConstructor()
```

(Test names should indicate _behavior_. :smile:)

File: test/Router/Zf2Test.php [hunk 2]
```diff
@@ -26,6 +27,15 @@ public function getRouter()
         return new Zf2Router($this->zf2Router->reveal());
     }
 
+    public function testGetRouterPropertyWithNullInConstruct()
```
Comment (weierophinney) [inline test/Router/Zf2Test.php]: Apply similar naming here.

**Comment by samsonasik** [comment]
@weierophinney done ;). I've renamed the test functions

File: test/Router/Zf2Test.php [hunk 3]
```diff
@@ -26,6 +27,15 @@ public function getRouter()
         return new Zf2Router($this->zf2Router->reveal());
     }
 
+    public function testWillLazyInstantiateAFastRouteCollectorIfNoneIsProvidedToConstructor()
```
Comment (weierophinney) [inline test/Router/Zf2Test.php]: This one should `s/FastRouteCollector/Zf2TreeRouteStack/`.

Comment (samsonasik) [inline test/Router/Zf2Test.php]: done ;)

**Comment by samsonasik** [comment]
should be merge-able now ;)

**Comment by weierophinney** [comment]
I realized during merge that the reflection work is something that PHPUnit's `assertAttributeInstanceOf()` already does, so I simplified the tests a bit. :)
